### ISSUE TITLE
Create a mechanism to bring hardware reference designs to the project

### ISSUE THREAD

**Issue by shaddi** [magma/magma#6865]
# Background

Many, if not all, Magma deployments require hardware. For example, the AGW is (typically) an on-path device that is deployed at or near a tower site for an operator. An operator could choose to use generic, commercially-available servers for this deployment scenario, but purpose-built hardware designed for this scenario could be helpful for ensuring low-cost and reliable deployments. 

Members of the Magma community have expressed interest in designing hardware specifically designed for Magma, but as of today there's not a way to "contribute" such designs to the Magma community and ensure they have visibility within the community.  

# Proposal

[Let's create a process by which Magma community members can contribute high-quality reference designs to the project.
](https://docs.google.com/presentation/d/1qhwK4Mz0NmX8zofR28NaliBHqDLCHhdtuNySD9ThaCM/edit?usp=sharing)

The main idea is that reference designs can be proposed by anyone, and, if approved, will be listed as a reference design for the project and on the project web page. Contributors would need to provide a description of the design, its specifications, intended use case, ordering information, and test results showing their design can actually run Magma. These proposals would be reviewed and accepted just like any other design proposal, with review of the proposal focused on completeness of the design proposal rather than the quality of the reference design itself. More details are in the linked slide deck.

**Comment by shaddi** [comment]
Feedback from the TSC (@hcgatewood please correct me if I'm getting this wrong!):

1. Consider a strategy for handling abandoned designs
2. Develop a proposal for who could be good codeowners for hardware reference designs, and consider a shepherding process for new hardware proposals as well

**Comment by shaddi** [comment]
I've updated the proposal as we discussed.

1. I recommend that reference design contributors be responsible for profiling their designs on each new release of Magma; doing this will give the community a sense of what designs are still maintained and will "auto deprecate" designs that are no longer being maintained by their contributors.
2. I have included an initial list of four folks for shepherding/reviewing designs: Tariq, Sudhi, and Vivek M from FBC and Alex B from FreedomFi. All have confirmed they are willing to serve in this capacity.
